Mathlib/Algebra/Order/Hom/Basic.lean

```diff
@@ -122,3 +166,201 @@ theorem le_map_div_add_map_div [Group α] [AddCommSemigroup β] [LE β] [MulLEAd
 -- #align tactic.positivity_map tactic.positivity_map
 
 --end Mathlib.Meta.Positivity
+
+/-! ### Group (semi)norms -/
+
+
+/-- `AddGroupSeminormClass F α` states that `F` is a type of `β`-valued seminorms on the additive
+group `α`.
+
+You should extend this class when you extend `AddGroupSeminorm`. -/
+class AddGroupSeminormClass (F : Type _) (α β : outParam <| Type _) [AddGroup α]
+  [OrderedAddCommMonoid β] extends SubadditiveHomClass F α β where
+  /-- The image of zero is zero. -/
+  map_zero (f : F) : f 0 = 0
+  /-- The map is invariant under negation of its argument. -/
+  map_neg_eq_map (f : F) (a : α) : f (-a) = f a
+#align add_group_seminorm_class AddGroupSeminormClass
+
+/-- `GroupSeminormClass F α` states that `F` is a type of `β`-valued seminorms on the group `α`.
+
+You should extend this class when you extend `GroupSeminorm`. -/
+@[to_additive]
+class GroupSeminormClass (F : Type _) (α β : outParam <| Type _) [Group α]
+  [OrderedAddCommMonoid β] extends MulLEAddHomClass F α β where
+  /-- The image of one is zero. -/
+  map_one_eq_zero (f : F) : f 1 = 0
+  /-- The map is invariant under inversion of its argument. -/
+  map_inv_eq_map (f : F) (a : α) : f a⁻¹ = f a
+#align group_seminorm_class GroupSeminormClass
+
+/-- `AddGroupNormClass F α` states that `F` is a type of `β`-valued norms on the additive group
+`α`.
+
+You should extend this class when you extend `AddGroupNorm`. -/
+class AddGroupNormClass (F : Type _) (α β : outParam <| Type _) [AddGroup α]
+  [OrderedAddCommMonoid β] extends AddGroupSeminormClass F α β where
+  /-- The argument is zero if its image under the map is zero. -/
+  eq_zero_of_map_eq_zero (f : F) {a : α} : f a = 0 → a = 0
+#align add_group_norm_class AddGroupNormClass
+
+/-- `GroupNormClass F α` states that `F` is a type of `β`-valued norms on the group `α`.
+
+You should extend this class when you extend `GroupNorm`. -/
+@[to_additive]
+class GroupNormClass (F : Type _) (α β : outParam <| Type _) [Group α]
+  [OrderedAddCommMonoid β] extends GroupSeminormClass F α β where
+  /-- The argument is one if its image under the map is zero. -/
+  eq_one_of_map_eq_zero (f : F) {a : α} : f a = 0 → a = 1
+#align group_norm_class GroupNormClass
+
+export AddGroupSeminormClass (map_neg_eq_map)
+
+export GroupSeminormClass (map_one_eq_zero map_inv_eq_map)
+
+export AddGroupNormClass (eq_zero_of_map_eq_zero)
+
+export GroupNormClass (eq_one_of_map_eq_zero)
+
+attribute [simp] map_one_eq_zero -- porting note: `to_additive` translation already exists
+
+attribute [simp] map_neg_eq_map
+
+attribute [simp] map_inv_eq_map -- porting note: `to_additive` translation already exists
```

Comment:
@YaelDillies, do you know what's going on here? Were the attributes in error in Lean 3?

Reply:
`to_additive` in mathlib3 doesn't throw an error when the declaration already exists. That might be it?

Reply:
So this was wrong in ml3?

Reply:
No, it's expected behavior. I think it's just that the syntax changed to `to_additive existing the_existing_lemma`.

Reply:
Does that mean we need to add `to_additive existing` in this PR?